### PR TITLE
Unit Tests for trainer and mar generation component

### PR DESCRIPTION
**Pull request by shrinath-suresh**
Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

The PR contains

Iris Classification example for unit test
Unit Test for Mar Generation component

### CODE REVIEW

File: pytorch_pipeline/components/trainer/component.py [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
+Calls the Executor for the PyTorch Lightning training to start."""
 import inspect
 import importlib
 from typing import Optional, Dict
-from pytorch_pipeline.components.base.base_component import BaseComponent
-from pytorch_pipeline.components.trainer.generic_executor import GenericExecutor
 from pytorch_pipeline.components.trainer.executor import Executor
 
 
-class Trainer(BaseComponent):
```
Comment (chauhang) [inline pytorch_pipeline/components/trainer/component.py:9]: @shrinath-suresh What is the reason removing the BaseComponent?

File: pytorch_pipeline/components/trainer/executor.py [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
+
+import os
+from argparse import Namespace
 import pytorch_lightning as pl
 import torch
-import os
-from pytorch_pipeline.components.trainer.generic_executor import GenericExecutor
 
 
-class Executor(GenericExecutor):
```
Comment (chauhang) [inline pytorch_pipeline/components/trainer/executor.py:7]: @shrinath-suresh What is the reason removing the GenericExecutor. These will be needed for common elements s we add more components

File: pytorch_pipeline/examples/cifar10/cifar10_pytorch.py [hunk 3]
```diff
@@ -114,7 +113,7 @@
 trainer = Trainer(
     module_file="cifar10_train.py",
     data_module_file="cifar10_datamodule.py",
-    module_file_args=parser,
```
Comment (chauhang) [inline pytorch_pipeline/examples/cifar10/cifar10_pytorch.py]: @shrinath-suresh Have we verified this change?

**Review by chauhang — CHANGES_REQUESTED** [review]
Please see comments inline

File: pytorch_pipeline/examples/cifar10/cifar10_pytorch.py (hunk 3 above)
Comment (shrinath-suresh) [reply pytorch_pipeline/examples/cifar10/cifar10_pytorch.py]: Yes Geeta. All the other arguments are of type `dict`. Only the module_file_args is passed as `ArgumentParser`. I changed it to `dict` to keep it uniform. 

And also ran the scripts locally to verify the examples.

**Review by chauhang — APPROVED** [review]
Approving for now. Need follow-on PR to add back the Base components

**Pull request merged**